Improve the MacOS documentation (#54)
maplibre/maplibre-rs: docs/src/development-guide/building.md

# Building for various Platforms

## Desktop

The build for desktop is very simple. You just have to run the following:

```bash
cargo build -p maplibre-demo
```

> You can use the *--release* parameter if you want to be in release mode instead of debug.

Reviewer: ```suggestion
You can use the `--release` parameter if you want to be in release mode instead of debug.
```

If you want to run the application:

```bash
cargo run -p maplibre-demo
```

> > __Note__: Make sure you have selected the right toolchain target within Rustup. You can use `rustup show` to see your

Reviewer: ```suggestion
> __Note__: Make sure you have selected the right toolchain target within rustup. You can use `rustup show` to see your
```

> > active toolchain. If you want to change the target of the build manually, use the cargo *--target* parameter.

Reviewer: ```suggestion
> active toolchain. If you want to change the target of the build manually, use the cargo `--target` parameter.
```

You can then run the app on an iOS Simulator or a real device. During the Xcode build process cargo is used to build
a static library for the required architecture.

## MacOS

You can build Unix Executables for MacOS, as explained in the first section, with one of the following toolchain targets:

* **x86_64-apple-darwin** for Intel's x86-64 processors
* **aarch64-apple-darwin** for ARM64 processors

Use one of the following commands to build for your desired target architecture:

```bash
cargo build -p maplibre-demo --target x86_64-apple-darwin
cargo build -p maplibre-demo --target aarch64-apple-darwin
```

If you want to build a proper MacOS application (in OSX terminology), you will need to use the XCode project
in the folder `./apple/xcode/`.

> After installing [XCode](https://apps.apple.com/us/app/xcode/id497799835?ls=1&mt=12) and [Rustup](https://rustup.rs/),

Reviewer: ```suggestion
After installing [XCode](https://apps.apple.com/us/app/xcode/id497799835?ls=1&mt=12) and [rustup](https://rustup.rs/),
```

build the Rust library inside `./apple` with Cargo:

```bash
cargo build -p apple --lib

> ```

Reviewer: This is no longer needed, but maybe we should point out that the XCode project is executing tomething like that using the correct `--target` flag.

Author: I have removed this part and I have added a little text to mention that :

> The XCode project is configured to automatically compile the Rust library with the correct target in the *Cargo Build* build phases configuration.

> Then open the project from the folder `./apple/xcode` with XCode. Select the scheme called *example(macOs)* and

Reviewer: ```suggestion
Then open the project from the folder `./apple/xcode` with XCode. Select the scheme called *example (macOS)* and
```

> After installing [XCode](https://apps.apple.com/us/app/xcode/id497799835?ls=1&mt=12) and [Rustup](https://rustup.rs/),
> build the Rust library inside `./apple` with Cargo:
> 
> ```bash
> cargo build -p apple --lib
> ```
> 
> Then open the project from the folder `./apple/xcode` with XCode. Select the scheme called *example(macOs)* and
> click on *Product -> Build for -> Running*. This will build the MacOS application for the version of OSX defined
> in the Build Settings.

Author: ```suggestion
Install [XCode](https://apps.apple.com/us/app/xcode/id497799835?ls=1&mt=12) and [rustup](https://rustup.rs/).
Then open the project from the folder `./apple/xcode` with XCode. Select the scheme called *example(macOS)* and
click on *Product -> Build for -> Running*. This will build the MacOS application for the version of OSX defined
in the Build Settings. The XCode project is configured to automatically compile the Rust library with the correct target
in the *Cargo Build* build phases configuration.
```
